utils/build_profiler_report.py

```diff
@@ -0,0 +1,166 @@
+            new_tag = ''
+            if BaselineDict and target not in BaselineDict.keys():
+                new_tag = '_NEW'
+
+            fp.write(str(CurrentDict[target]) + ' ' + target
+                     + '_' + FormatTime(CurrentDict[target]) + new_tag + '\n')
+
+
+def GetTotalTime(TargetDurationDict: dict) -> int:
+    """Return sum of durations for all targets in dict"""
+    total_time = 0
+    for target in TargetDurationDict.keys():
+        total_time += TargetDurationDict[target]
+    return total_time
+
+
+def GetTotalTimeIntersect(TargetDurationDict_A: dict, TargetDurationDict_B: dict) -> int:
+    """Return sum of durations for targets in A that are also in B"""
+    total_time = 0
+    for target in TargetDurationDict_A.keys():
+        if target in TargetDurationDict_B.keys():
+            total_time += TargetDurationDict_A[target]
+    return total_time
+
+
+def PrintReport(CurrentDict: dict, BaselineDict: dict = None) -> None:
+    """Print report with results of profiling to stdout"""
+
+    # If the targets have changed between baseline and current, we are using total_time_intersect
+    # to calculate delta (ratio of durations of targets) instead of total_time
+    if BaselineDict and BaselineDict.keys() != CurrentDict.keys():
+        print("Warning: the targets in the current logfile differ from those in the baseline\
+             logfile; therefore the Δ delta is calculated without taking the added/removed\
+                  targets into consideration. If the added/removed targets modify the behavior of\
+                      targets in both logfiles, the Δ delta may not make sense.\n-----")
+        target_mismatch = True
+        total_time_current_intersect = GetTotalTimeIntersect(CurrentDict, BaselineDict)
+        total_time_baseline_intesect = GetTotalTimeIntersect(BaselineDict, CurrentDict)
+    else:
+        target_mismatch = False
+
+    header = [f'{"Target:":60}', f"{'%':4}", f"{'Δ':5}", f"{'T':8}", f"{'%Δ':5}", "Note"]
+    print(' | '.join(header))
+
+    total_time_current = GetTotalTime(CurrentDict)
+    if BaselineDict:
+        total_time_baseline = GetTotalTime(BaselineDict)
+
+    # sort targets by % taken of build time
+    temp = {k: v for k, v in sorted(CurrentDict.items(), key=lambda item: item[1], reverse=True)}
+    CurrentDict = temp
+
+    for target in CurrentDict.keys():
+        # percentage of build time that the target took
+        perc = CurrentDict[target]/total_time_current * 100
+
+        # difference between perc in current and in baseline
+        delta = 0
+        if BaselineDict:
+            if target_mismatch:
+                if target in BaselineDict.keys():
+                    delta = CurrentDict[target]/total_time_current_intersect * 100 - \
+                        BaselineDict[target]/total_time_baseline_intesect * 100
+
+            else:
+                delta = perc - (BaselineDict[target]/total_time_baseline * 100)
+
```

Comment:
Try to extract this to a function.

Reply:
I don't see the reason for moving this into a function, considering the fact that I would have to pass 7 variables to such function. That would just make the code less readable in my opinion.